### PR TITLE
Add acceleration structure meta-commands

### PR DESCRIPTION
**Pull request by fabian-lunarg**
- Extend capture/tracking to cover acceleration-structures (AS) and dependencies
- Add meta-commands to rebuild AS during replay
- create all stubs in replay, implement in follow-up

this PR belongs to [portable raytracing initiative](https://github.com/LunarG/gfxreconstruct/issues/1526)

### CODE REVIEW

**Review by MarkY-LunarG — APPROVED** [review]

**Comment by MarkY-LunarG** [comment]
Overall looks good to me.  I did see you used the reserve bits, are you sure you can use those?  I thought they were for something else.

File: framework/decode/vulkan_replay_consumer_base.h [hunk 1]
```diff
@@ -189,16 +189,29 @@ class VulkanReplayConsumerBase : public VulkanConsumer
                                                                 format::HandleId   commandBuffer,
                                                                 StructPointerDecoder<Decoded_VkPushDescriptorSetWithTemplateInfoKHR>*
                                                                     pPushDescriptorSetWithTemplateInfo) override;
-    virtual void Process_vkCreateRayTracingPipelinesKHR(
```
[the remark this answers is not in the document]
Comment (fabian-lunarg) [reply framework/decode/vulkan_replay_consumer_base.h:192]: clang-tidy complains when having both, so for new code I'd only use `override`, it's also less typing :) here I might have used the auto-correct hotkey in my IDE while adding code next to it, was not really required.

**Comment by fabian-lunarg** [comment]
> Overall looks good to me. I did see you used the reserve bits, are you sure you can use those? I thought they were for something else.

thanks @MarkY-LunarG ! regarding the reserved enum-values. I first created new values but then realized those bits were reserved by @bartosz-muszarski-arm. they also matched the values used in his [PR](https://github.com/LunarG/gfxreconstruct/pull/1533)  - it all lined up

File: framework/encode/vulkan_device_address_tracker.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+#include "encode/vulkan_device_address_tracker.h"
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(encode)
+
+void encode::VulkanDeviceAddressTracker::TrackBuffer(const vulkan_wrappers::BufferWrapper* wrapper)
+{
+    if (wrapper != nullptr && wrapper->handle != VK_NULL_HANDLE && wrapper->address != 0 && wrapper->created_size != 0)
+    {
+        _buffer_addresses[wrapper->address] = { wrapper->handle, wrapper->created_size };
```
[the remark this answers is not in the document]
Comment (fabian-lunarg) [reply framework/encode/vulkan_device_address_tracker.cpp]: let me check

**Comment by bradgrantham-lunarg** [comment]
@fabian-lunarg if @MarkY-LunarG has approved then that's sufficient.

File: framework/encode/vulkan_device_address_tracker.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (fabian-lunarg) [reply framework/encode/vulkan_device_address_tracker.cpp]: no, not required here. we only need address+size for a range-based lookup to retrieve a `VkBuffer`-handle.
for anything else we go -> BufferWrapper -> usage/whatever.

File: framework/decode/vulkan_replay_consumer_base.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (fabian-lunarg) [reply framework/decode/vulkan_replay_consumer_base.h:192]: at some point we 'could' sanitize this in a dedicated PR, e.g. via:
https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-override.html

**Comment by MarkY-LunarG** [comment]
> > Overall looks good to me. I did see you used the reserve bits, are you sure you can use those? I thought they were for something else.
> 
> thanks @MarkY-LunarG ! regarding the reserved enum-values. I first created new values but then realized those bits were reserved by @bartosz-muszarski-arm. they also matched the values used in his [PR](https://github.com/LunarG/gfxreconstruct/pull/1533) - it all lined up

Thank you for verifying!

**Pull request merged**